python/race_model/EloRaceModel.py

```diff
@@ -41,30 +42,41 @@ def getExpectedScore(self, a, b):
         '''Returns a's expected score against b. A float value between 0 and 1'''
         return 1 / (1 + 10 ** ((b - a) / 400))
 
-    def adjustEloRating(self, driverId, adjustment):
-        self.drivers[driverId].rating += (adjustment * K_FACTOR) # TODO check if this is correct
+    def adjustEloRating(self, driverId, adjustment, trackId):
+        self.drivers[driverId].rating += (adjustment * K_FACTOR)  # TODO check if this is correct
+        self.drivers[driverId].trackRatings[trackId] += (adjustment * K_FACTOR)
 
     def adjustEloRatingConstructor(self, constructor, adjustment):
         constructor.rating += (adjustment * K_FACTOR) # TODO check if this is correct
 
     def adjustEloRatingEngine(self, engine, adjustment):
-        engine.rating += (adjustment * K_FACTOR) # TODO check if this is correct
+        engine.rating += (adjustment * K_FACTOR)  # TODO check if this is correct
+
+    def _addNewTracksToEntities(self, driverId, trackId):
+        if trackId not in self.drivers[driverId].trackRatings[trackId]:
+            # TODO maybe change defaults
+            self.drivers[driverId].trackRatings[trackId] = 2200
+        if trackId not in self.drivers[driverId].constructor.trackRatings:
+            # TODO maybe change defaults
+            self.drivers[driverId].constructor.trackRatings[trackId] = 2200
+        if trackId not in self.drivers[driverId].constructor.engine.trackRatings:
+            # TODO maybe change defaults
+            self.drivers[driverId].constructor.engine.trackRatings[trackId] = 2200
+        
+    
         
 class EloConstructor:
     def __init__(self, name, engine):
         self.name = name
         self.engine = engine
-        self.rating = 2200  # Default rating
```

Comment:
We shouldn't delete the `rating` field constructors and engines, but add the `trackRatings` field alongside it